PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+
+    SpreadsheetApp.flush();
+    // STEP 4: If there are more URLs to run, cycle through this function again 
+    toast("Done","Batch finished")
+    if(batch.length >= NUMBER_OF_URLS_PER_BATCH){
+        return true;
+    }
+    // if not, we exit and finish the program
+    removeBatchTriggers();
+    return false;
+}
+
+/**
+ * To parse the results from the PSI API, we rely on the tab "Fields" that are going to contain the functions to run
+ */
+function parseResults(api, fetch, batch, sheet, save_screenshot = false){
+    // STEP 1: We retrieve the fields we want to extract from the JSON from the tab "Fields"
+    // They are made of the following elements:
+    // 1. A "Method", which corresponds to the current API being run (here we will focus only on PSI API)
+    // 2. A "Field" which is the name of the field calculated and will match the "Results" tab
+    // 3. A "Data" which contains the way to access the information from the "content" field from the response
+    // Example: "LH Version" reads `content["lighthouseResult"]["lighthouseVersion"]` to read the Lighthouse version from the API
+    let sheet_fields = FIELD_SHEET;
+    let sheet_fields_last_row = sheet_fields.getLastRow();
+    let fields = sheet_fields.getRange(1, 1, sheet_fields_last_row, 3).getValues();
+    fields = fields.map(d=>{
+        return {
+            "method": d[0],
+            "field": d[1],
+            "data": d[2]
+        }
+    })
+    // We only focus on the fields matching the PSI API method
+    if(api == "PSI"){fields = fields.filter(d=>d.method == "PSI API")}
+    if(api == "CrUX"){fields = fields.filter(d=>d.method == "CrUX")}
+    if(api == "CrUX History"){fields = fields.filter(d=>d.method == "CrUX History")}
+    if(api == "Green Domain"){fields = fields.filter(d=>d.method == "Green Domain")}
+    if(api == "Accessibility"){fields = fields.filter(d=>d.method == "Accessibility")}
+    let green_domains = [];
+    if(api == "Sustainability"){
+      fields = fields.filter(d=>d.method == "Sustainability")
+      green_domains = SpreadsheetApp
+                        .getActiveSpreadsheet()
+                        .getSheetByName("Green Domains (GWF)")
+                        .getRange("F2:J")
+                        .getValues()
+                        .filter(d=>d[4]==true)
+                        .map(d=>d[0])
+      Logger.log("GREEN DOMAINS")
+      Logger.log(green_domains)  
+    }
+
+
+    // STEP 2: We iterate through all the fields
+    for (let i = 0; i < fetch.length; i++) {
+        let fields_for_debug = [new Date().toISOString().slice(0, 10), batch[i].label, batch[i].url, batch[i].device, batch[i].origin];
+        let debug = "";
+        // Indicate that we worked on that row in the Config tab
+        sheet.getRange(batch[i].id+1,COLUMN_STATUS+1).setValue("✅")
+        // Read and parse the data for that specific cakk
+        const data = fetch[i];
+        Logger.log(data)
```

Comment:
Log message could use more context.

Reply:
legacy debug purposes here! I removed it